pymc/distributions/multivariate.py

```diff
@@ -572,8 +572,15 @@ def moment(rv, size, n, p):
         n = pt.shape_padright(n)
         mode = pt.round(n * p)
         diff = n - pt.sum(mode, axis=-1, keepdims=True)
-        inc_bool_arr = pt.abs(diff) > 0
-        mode = pt.inc_subtensor(mode[inc_bool_arr.nonzero()], diff[inc_bool_arr.nonzero()])
+        if len(mode.shape.eval()) == 1:
+            mode = mode.reshape((1, -1))
+        for i in range(mode.shape.eval()[0]):
+            elem_to_change = pt.abs(diff[i].astype(int))
```

Comment:
Instead of trying to distribute the difference among the first n largest elements, we could try to assign it all to the largest elements of each mode.

That is still bound to fail now and then but perhaps less often than the even dumber approach of distributing it to the first element of each mode.

That way the algorithm would still work in a vectorized manner.

Even your algorithm I think is not bullet proof, e.g. if the only solution is to move the largest entry in mode by more than one entry.

Reply:
@ricardoV94 I like the idea  to assign all difference to the largest element of each mode, will implement it.